### PR TITLE
refactor(galaxy): refactor ansible galaxy to a class

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes:

Migrate the Ansible Galaxy datasource to a class-based datasource

Additionally, applies the auto-caching decorator, and extracts the JSON return type out into a `GalaxyResult` type.

## Context:

Related to #8647.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by zharinov — APPROVED** [review]
lgtm

**Review by rarkins — APPROVED** [review]

File: lib/datasource/galaxy/index.ts [hunk 1]
```diff
@@ -1,71 +1,75 @@
 import { logger } from '../../logger';
-import { ExternalHostError } from '../../types/errors/external-host-error';
-import * as packageCache from '../../util/cache/package';
-import { Http } from '../../util/http';
+import { cache } from '../../util/cache/package/decorator';
+import { HttpResponse } from '../../util/http';
```
Comment (viceice) [inline lib/datasource/galaxy/index.ts:3]:
```suggestion
import type { HttpResponse } from '../../util/http';
```

File: lib/datasource/galaxy/index.ts [hunk 2]
```diff
@@ -1,71 +1,75 @@
...
-  try {
-    let res: any = await http.get(galaxyAPIUrl);
-    if (!res || !res.body) {
+  readonly customRegistrySupport = false;
+
+  readonly defaultRegistryUrls = ['https://galaxy.ansible.com/'];
+
+  @cache({
+    namespace: 'datasource-galaxy',
+    key: (getReleasesConfig: GetReleasesConfig) => getReleasesConfig.lookupName,
```
Comment (viceice) [inline lib/datasource/galaxy/index.ts:21]:
```suggestion
    key: ({ lookupName }: GetReleasesConfig) => lookupName,
```
Just two small things. 😉

**Comment by JamieMagee** [comment]
@viceice I'll send out a follow-up PR to address your comments 😄

**Pull request merged**